front/app/api/projects/useResetProject.ts

```diff
@@ -0,0 +1,38 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+import adminPublicationsKeys from 'api/admin_publications/keys';
+import adminPublicationsStatusCountsKeys from 'api/admin_publications_status_counts/keys';
+import eventsKeys from 'api/events/keys';
+import ideasKeys from 'api/ideas/keys';
+
+import fetcher from 'utils/cl-react-query/fetcher';
+
+import projectsKeys from './keys';
+
+const resetProject = (id: string) =>
+  fetcher({
+    path: `/projects/${id}/reset_participation_data`,
+    action: 'delete',
+  });
+
+const useResetProject = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: resetProject,
+    onSuccess: (_data, id) => {
+      queryClient.invalidateQueries({ queryKey: projectsKeys.lists() });
+      queryClient.invalidateQueries({ queryKey: projectsKeys.item({ id }) });
+      queryClient.invalidateQueries({
+        queryKey: adminPublicationsKeys.lists(),
+      });
+      queryClient.invalidateQueries({
+        queryKey: adminPublicationsStatusCountsKeys.items(),
+      });
+      queryClient.invalidateQueries({ queryKey: ideasKeys.lists() });
+      queryClient.invalidateQueries({ queryKey: eventsKeys.lists() });
+    },
```

Comment:
In this case, I think you are right, we don't need to invalidate the admin publications because in fact, this action cannot affect them. I added them because they are included in the regular delete action but there's no need. We also don't have to invalidate the project list in this case (same reasoning).

I did add invalidation for the new items we are resetting with voting - baskets, comments, etc.